src/placeos-core/resource.cr

```diff
@@ -87,14 +88,16 @@ abstract class PlaceOS::Core::Resource(T)
 
   # Load all resources from the database, push into a channel
   #
-  private def load_resources
-    count = 0
-    T.all.each do |resource|
-      event_channel.send({resource: resource, action: Action::Created})
-      count += 1
+  private def load_resources : Nil
+    waiting = [] of Promise::DeferredPromise(Nil)
+    T.all(runopts: {"read_mode" => "majority"}).in_groups_of(channel_buffer_size).each do |resources|
+      resources.each do |resource|
+        next unless resource
+        waiting << Promise.defer { _process_event({resource: resource.not_nil!, action: Action::Created}) }
+      end
+      Promise.all(waiting).get
+      waiting.clear
     end
-
-    count
```

Comment:
It would be good to have an `info` or `debug` level Log here to note how many resources were loaded